File: app/src/main/java/com/app/rupiksha/extra/SoftKeyboard.java
package com.app.rupiksha.extra;
//
//import android.annotation.SuppressLint;
//import android.app.Activity;
//import android.content.Context;
//import android.os.Handler;
//import android.os.Looper;
//import android.view.MotionEvent;
//import android.view.View;
//import android.view.View.OnTouchListener;
//import android.view.ViewGroup;
//import android.view.inputmethod.InputMethodManager;
//import android.widget.EditText;
//
//public class SoftKeyboard
//{
//    private static Activity mActivity;
//
//    public SoftKeyboard(Activity activity) {
//        this.mActivity = activity;
//    }
//    public void setupUI(final View view) {
//        try {
//            if (!(view instanceof EditText)) {
//                view.setOnTouchListener(new OnTouchListener() {
//                    @Override
//                    @SuppressLint("ClickableViewAccessibility")
//                    public boolean onTouch(View v, MotionEvent event) {
//                        hideSoftKeyboard();
//                        hideSoftKeyboardDialog(view);
//                        return false;
//                    }
//                });
//            }
//
//            if (view instanceof ViewGroup) {
//                for (int i = 0; i < ((ViewGroup) view).getChildCount(); i++) {
//                    View innerView = ((ViewGroup) view).getChildAt(i);
//                    setupUI(innerView);
//                }
//            }
//        } catch (Exception e) {
//            e.printStackTrace();
//        }
//    }
//    public void show_keyboard() {
//        new Handler(Looper.getMainLooper()).postDelayed(new Runnable() {
//            @Override
//            public void run() {
//                InputMethodManager imm = (InputMethodManager) mActivity.getSystemService(Activity.INPUT_METHOD_SERVICE);
//                imm.toggleSoftInput(InputMethodManager.SHOW_FORCED, 0);
//            }
//        },500);
//    }
//    public void hideSoftKeyboardDialog(View view) {
//        try {
//            view.requestFocus();
//            view.postDelayed(() -> {
//                InputMethodManager keyboard = (InputMethodManager)
//                        mActivity.getSystemService(Context.INPUT_METHOD_SERVICE);
//                if (false) {
//                    keyboard.showSoftInput(view, 0);
//                }else{
//                    keyboard.hideSoftInputFromWindow(view.getWindowToken(), 0);
//                }
//            }, 200);
//        } catch (Exception e) {
//            e.printStackTrace();
//        }
//    }
//        public static void hideSoftKeyboard() {
//        try {
//            InputMethodManager inputMethodManager = (InputMethodManager) mActivity
//                    .getSystemService(Context.INPUT_METHOD_SERVICE);
//            inputMethodManager.hideSoftInputFromWindow(mActivity
//                    .getCurrentFocus().getWindowToken(), 0);
//        } catch (Exception e) {
//            e.printStackTrace();
//        }
//    }
//
//    public static void hidekeyboard(Context context){
//        InputMethodManager imm1 = (InputMethodManager) context.getSystemService(Context.INPUT_METHOD_SERVICE);
//        assert imm1 != null;
//        imm1.toggleSoftInput(InputMethodManager.SHOW_IMPLICIT, 0);
//    }
//
//
//    //Show-Hide
//    public void edtRequestKeyboard(final View view, final boolean isOpen) {
//
//            view.requestFocus();
//            view.postDelayed(new Runnable() {
//                @Override
//                public void run() {
//                    InputMethodManager keyboard = (InputMethodManager)
//                            mActivity.getSystemService(Context.INPUT_METHOD_SERVICE);
//                    if (isOpen) {
//                            keyboard.showSoftInput(view, 0);
//                    } else {
//                            keyboard.hideSoftInputFromWindow(view.getWindowToken(), 0);
//                    }
//                }
//            }, 200);
//    }
//}

File: app/src/main/java/com/app/rupiksha/extra/SoftKeyboard.kt
package com.app.rupiksha.extra

import android.app.Activity
import android.content.Context
import android.os.Handler
import android.os.Looper
import android.view.View
import android.view.ViewGroup
import android.view.inputmethod.InputMethodManager
import android.widget.EditText

class SoftKeyboard(private val activity: Activity) {

    fun setupUI(view: View) {
        try {
            if (view !is EditText) {
                view.setOnTouchListener { _, _ ->
                    hideSoftKeyboard()
                    hideSoftKeyboardFromView(view)
                    false
                }
            }

            if (view is ViewGroup) {
                for (i in 0 until view.childCount) {
                    val innerView = view.getChildAt(i)
                    setupUI(innerView)
                }
            }
        } catch (e: Exception) {
            e.printStackTrace()
        }
    }

    fun showKeyboard() {
        Handler(Looper.getMainLooper()).postDelayed({
            val imm = activity.getSystemService(Context.INPUT_METHOD_SERVICE) as InputMethodManager
            imm.toggleSoftInput(InputMethodManager.SHOW_FORCED, 0)
        }, 500)
    }

    fun hideSoftKeyboardFromView(view: View) {
        try {
            view.requestFocus()
            view.postDelayed({
                val keyboard =
                    activity.getSystemService(Context.INPUT_METHOD_SERVICE) as InputMethodManager
                keyboard.hideSoftInputFromWindow(view.windowToken, 0)
            }, 200)
        } catch (e: Exception) {
            e.printStackTrace()
        }
    }

    fun hideSoftKeyboard() {
        try {
            val inputMethodManager =
                activity.getSystemService(Context.INPUT_METHOD_SERVICE) as InputMethodManager
            activity.currentFocus?.let {
                inputMethodManager.hideSoftInputFromWindow(it.windowToken, 0)
            }
        } catch (e: Exception) {
            e.printStackTrace()
        }
    }

    companion object {
        fun hideKeyboard(context: Context) {
            val imm = context.getSystemService(Context.INPUT_METHOD_SERVICE) as InputMethodManager
            imm.toggleSoftInput(InputMethodManager.SHOW_IMPLICIT, 0)
        }
    }

    fun edtRequestKeyboard(view: View, isOpen: Boolean) {
        view.requestFocus()
        view.postDelayed({
            val keyboard =
                activity.getSystemService(Context.INPUT_METHOD_SERVICE) as InputMethodManager
            if (isOpen) {
                keyboard.showSoftInput(view, 0)
            } else {
                keyboard.hideSoftInputFromWindow(view.windowToken, 0)
            }
        }, 200)
    }
}
